pdal/GDALUtils.cpp

```diff
@@ -774,6 +777,107 @@ OGRGeometry *createFromGeoJson(const std::string& s, std::string& srs)
     return newGeom;
 }
 
+
+std::vector<OGRGeometry*> fetchOGRGeometries(const NL::json ogr)
+{
+    const NL::json& layer = ogr.at("layer");
```

Comment:
>  there's no schema and there's no checking here for anything going wrong

This code throws if the expected keys are not found. Keys that are optional are handled. It happens at runtime. There's a bunch of stuff we don't know about the given datasource until we try to open it. Opening OGR datasources can be expensive. We don't want to do it multiple times if we can avoid it.

We have no way in ProgramArgs to handle stuff like this `ogr` option. It doesn't make sense to flatten it out into 20 different options on a stage, and we also want to add this capability to other stages like `filters.crop`.

Reply:
I'm not suggesting that we do something other than JSON.  I'm suggesting that the JSON input be checked for errors.

Reply:
Check for errors at what time? `initialize()` is when it is happening right now.

Reply:
That's a fine time to check the pipeline input.

I think it would be better to do something like build the query in initialize() and then use it in ready().  But I'd have to look at the whole reader a bit to be sure what's feasible and reasonable.

My concern is just that the input validation occur at some point.  Right now it's very loose, which is what tends to happen with this JSON stuff.  I blame at least a couple of days' time with 3D tiles on this.